injectproxy/rules.go

```diff
@@ -185,3 +173,54 @@ func (r *routes) rules(resp *http.Response) error {
 
 	return nil
 }
+
+func (r *routes) rules(resp *http.Response) error {
+	return modifyAPIResponse(resp, func(b []byte) (interface{}, error) {
```

Comment:
just a small nit: I really liked the simple idiom of the `rules(resp *http.Response)` modifier. Instead of converting a `http response` modifier here, we could modify `modifyAPIResponse` in a way that it converts a `func(*http.Response) error` into a `func(*apiResponse) error` .

Then we could have a nice and easy `func (r *routes) rules(resp *apiResponse) error` signature here without passing callbacks.

I have a proposal in a branch, let me know if this is a bit simpler, I will past a commit in a second.

Reply:
@simonpasquier https://github.com/openshift/prom-label-proxy/commit/5809220166f1a0065e8716138b5ad61990ab9164